examples/containers/multi/helpers/driver.sh

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 Couchbase, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file  except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the  License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#########################
+# Pre-conditions:
+#   - The cbs_exp_image_name Docker image built (handled by the run.sh entrypoint)
+#   - A non-zero $NUM_NODES (a default is specified in the run.sh entrypoint)
+
+# Post-conditions:
+#   - $NUM_NODES Couchbase Server/exporter containers with Couchbase Server ready
+#     to receive requests, and the exporter actively exporting data
+function start_new_nodes() {
+
+    local num_nodes=$1
+    local nodes_ready=() 
+
+    echo "---- Starting $num_nodes Couchbase Server and Exporter nodes ----"
+
+    local i=0
+    for ((i; i<num_nodes; i++)); do
+        docker run -d --rm --name "node$i" --hostname="node$i.local" --network=multi_shared_network \
+        -p $((8091+i)):8091 "cbs_server_exp" > /dev/null
+        nodes_ready+=(false)
+    done
+
+    # Simple block until all nodes ready
+    echo "Waiting for nodes to come up..." && sleep "$num_nodes"
+    while true; do
+        local j=0
+        for ((j; j<num_nodes; j++)); do
+            if ! ${nodes_ready[$j]}; then
+                if docker exec "node$j" curl -fs localhost:8091 > /dev/null; then
+                    nodes_ready[$j]=true
+                    echo "Node $j ready!"
+                fi
+            fi
+        done
+
+        ready=true
+        for b in "${nodes_ready[@]}"; do if ! $b; then ready=false; fi; done
+        if $ready; then break; else echo "..." && sleep 5; fi
```

Comment:
Might be easier to read broken across multiple lines?
```suggestion
        for b in "${nodes_ready[@]}"; do
            if ! $b; then
                ready=false
            fi
        done
        if $ready; then
            break
        else
            echo "..."
            sleep 5
        fi
```

Reply:
Also use `echo -n .` to keep it on the same line